Dockerfile

```
@@ -33,5 +33,6 @@ ENV BUNDLE_PATH="$GEM_HOME" \
 
 # path recommendation: https://github.com/bundler/bundler/pull/6469#issuecomment-383235438
 ENV PATH $GEM_HOME/bin:$BUNDLE_PATH/gems/bin:/usr/lib/fullstaq-ruby/versions/${RUBY_VERSION}/bin:$PATH
+RUN echo -e "\nPATH=$GEM_HOME/bin:$BUNDLE_PATH/gems/bin:/usr/lib/fullstaq-ruby/versions/${RUBY_VERSION}/bin:$PATH" >> /etc/profile
```

You need to use single quotes, otherwise environment variables will be expanded and following `PATH` will be resulted:

```
PATH=/usr/local/bundle/bin:/usr/local/bundle/gems/bin:/usr/lib/fullstaq-ruby/versions/3.2.2-jemalloc/bin:/usr/local/bundle/bin:/usr/local/bundle/gems/bin:/usr/lib/fullstaq-ruby/versions/3.2.2-jemalloc/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
```

instead of expected:

```
/usr/local/bundle/bin:/usr/local/bundle/gems/bin:/usr/lib/fullstaq-ruby/versions/3.2.2-jemalloc/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
```

Also (this is a nitpick, but) it is more clean way to do it will be not appending `/etc/profile`, but creating new file in `/etc/profile.d/`.

So it should look like this:
```suggestion
RUN echo -e 'PATH=$GEM_HOME/bin:$BUNDLE_PATH/gems/bin:/usr/lib/fullstaq-ruby/versions/${RUBY_VERSION}/bin:$PATH' >> /etc/profile.d/fullstaq-ruby.sh
```

---

Thank you for the detailed explanation. I just learned today that there is such a problem with writing like this in dockerfile.

I didn't change it to single quotes, but adjusted the code order because echo doesn't parse variables inside single quotes.

---

echo doesn't expand environment variables, but shell does. And `RUN` command in this syntax is executed in a shell. See [`RUN` directive docs](https://docs.docker.com/engine/reference/builder/#run) for details.

> `RUN <command>` (_shell_ form, the command is run in a shell, which by default is `/bin/sh -c` on Linux or `cmd /S /C` on Windows)
> …
> it is the shell that is doing the environment variable expansion, not docker.

So you have to use single quotes here. I already changed it in the pull request.